infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LRRollingUpgradeHandler.java

```diff
@@ -81,5 +101,29 @@ public boolean isLRUpgradeInProgress(TxnContext txnContext) {
      */
     public void migrateData(TxnContext txnContext) {
         // Data migration to be added here.
+        addSnapshotSyncEventOnUpgradeCompletion(txnContext);
+    }
+
+    /**
+     * Add flag to event table to trigger snapshot sync.
+     */
+    public void addSnapshotSyncEventOnUpgradeCompletion(TxnContext txnContext) {
+        UUID rollingUpgradeForceSyncId = UUID.randomUUID();
+
+        // Write a rolling upgrade force snapshot sync event to the logReplicationEventTable
+        LogReplicationMetadata.ReplicationEventInfoKey key = LogReplicationMetadata.ReplicationEventInfoKey.newBuilder()
+                .build();
+
+        LogReplicationMetadata.ReplicationEvent event = LogReplicationMetadata.ReplicationEvent.newBuilder()
+                .setEventId(rollingUpgradeForceSyncId.toString())
+                .setType(LogReplicationMetadata.ReplicationEvent.ReplicationEventType.UPGRADE_COMPLETION_FORCE_SNAPSHOT_SYNC)
+                .setEventTimestamp(Timestamp.newBuilder().setSeconds(Instant.now().getEpochSecond()).build())
+                .build();
+
+        Table<LogReplicationMetadata.ReplicationEventInfoKey, LogReplicationMetadata.ReplicationEvent, Message> replicationEventTable =
+                txnContext.getTable(REPLICATION_EVENT_TABLE_NAME);
```

Comment:
For ITs this table can be opened outside, but in real LR jvm how do we ensure that this `REPLICATION_EVENT_TABLE_NAME` gets opened before this part of the code executes?
Because otherwise we will face an `IllegalArgumentException` right?

Reply:
Yes, we can include this change in #3605 since a corfuStore instance will be available to do an open table within the upgrade handler.

Reply:
CorfuStore is also being passed to LRRollingUpgradeHandler in this PR  https://github.com/CorfuDB/CorfuDB/pull/3596.

The table is being opened in the constructor there.